### ISSUE TITLE
BUG: Redis instrumentation crashes when using `client.call([:command, args])`

### ISSUE THREAD

**Issue by brafales** [DataDog/dd-trace-rb#796]
Hi,

We've encountered an issue when upgrading to version 0.25.1 of the gem. We have code in our application that uses the redis-rb `Redis` class `call` method by calling it the following way:

```ruby
Redis.call([:del, "key1", "key2", "key3"])
```

As per the docs: https://www.rubydoc.info/github/redis/redis-rb/Redis#call-instance_method.

As far as we know this is a valid way to interact with Redis, but raises an error in the following line: https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/redis/quantize.rb#L30 since `command_args.first` evaluates to an Array and therefore calling `to_sym` on an Array raises an error.

Here's a stack trace I can provide:
```
vendor/bundle/ruby/2.6.0/gems/ddtrace-0.26.0/lib/ddtrace/contrib/redis/quantize.rb:30:in `auth_command?': undefined method `to_sym' for #<Array:0x00007f573a3671b0>
Did you mean?  to_s
               to_set (NoMethodError)
    from vendor/bundle/ruby/2.6.0/gems/ddtrace-0.26.0/lib/ddtrace/contrib/redis/quantize.rb:23:in `format_command_args'
    from vendor/bundle/ruby/2.6.0/gems/ddtrace-0.26.0/lib/ddtrace/contrib/redis/patcher.rb:67:in `block in call'
    from vendor/bundle/ruby/2.6.0/gems/ddtrace-0.26.0/lib/ddtrace/tracer.rb:303:in `trace'
    from vendor/bundle/ruby/2.6.0/gems/ddtrace-0.26.0/lib/ddtrace/contrib/redis/patcher.rb:64:in `call'
    from vendor/bundle/ruby/2.6.0/gems/redis-4.1.2/lib/redis.rb:88:in `block in call'
    from vendor/bundle/ruby/2.6.0/gems/redis-4.1.2/lib/redis.rb:52:in `block in synchronize'
    from /usr/local/lib/ruby/2.6.0/monitor.rb:230:in `mon_synchronize'
    from vendor/bundle/ruby/2.6.0/gems/redis-4.1.2/lib/redis.rb:52:in `synchronize'
    from vendor/bundle/ruby/2.6.0/gems/redis-4.1.2/lib/redis.rb:87:in `call'
```

We have since changed our code to use `REDIS.del("key1", "key2")` instead but we think given the other command is expected to be valid, the gem should be prepared to process it.

To reproduce:
In any ddtrace enabled ruby project using redis instrumentation, open a ruby console and type: `Redis.call([:del, "key1", "key2", "key3"])` where `Redis` is a Redis client using the `redis-rb` gem.

Expected result:
- ddtracer should report metrics of a `DEL` command in Redis

Actual result:
Error is raised.

**Comment by delner** [comment]
Hey @brafales, sorry for the late response, but thank you for the thorough report! We will try to reproduce this on our end; the details you provided should make that pretty simple.

**Comment by dustMason** [comment]
We also just bumped into this bug while trying to upgrade. It seems that the bug was introduced by this change https://github.com/DataDog/dd-trace-rb/commit/083f52f26436c9ff7f99c60c016de9e15a7109f0#diff-ed19b6778e2d6d59ba37ecd23d0307bdR30, which doesn't account for `command_args` being an array containing a single array of redis command args.

**Comment by marcotc** [comment]
Hey @brafales @dustMason, we've just released version `0.28.0` with a [fix](https://github.com/DataDog/dd-trace-rb/pull/817) to this issue.

I'm going to close this issue for now, let us know if the problem persists.
